qualtran/bloqs/basic_gates/cnot.py

```diff
@@ -96,6 +99,27 @@ def add_my_tensors(
     def on_classical_vals(self, ctrl: int, target: int) -> Dict[str, 'ClassicalValT']:
         return {'ctrl': ctrl, 'target': (ctrl + target) % 2}
 
+    def get_ctrl_system(
+        self, ctrl_spec: Optional['CtrlSpec'] = None
+    ) -> Tuple['Bloq', 'AddControlledT']:
+        from qualtran.bloqs.basic_gates.toffoli import Toffoli
+
+        if ctrl_spec is None or ctrl_spec == CtrlSpec():
+            bloq = Toffoli()
+
+            def add_controlled(
+                bb: 'BloqBuilder', ctrl_soqs: Sequence['SoquetT'], in_soqs: Dict[str, 'SoquetT']
+            ) -> Tuple[Iterable['SoquetT'], Iterable['SoquetT']]:
+                (c2,) = ctrl_soqs
+                (c1, c2), target = bb.add(
+                    bloq, ctrl=np.array([in_soqs['ctrl'], c2]), target=in_soqs['target']
+                )
+                return (c1,), (c2, target)
+
+            return bloq, add_controlled
```

Comment:
It's worth noting that this introduces an implicit ordering between the wires that get plugged into the two soquets of the newly added Toffoli gate. We've faced this issue in the past where two circuits are equivalent but they won't compare equal if, for example, one has `Toffoli(c1, c2, target)` and other has `Toffoli(c2, c1, target)`. 

Special purpose code was added in Cirq to make this equality return `True`. I don't think we have any similar plans for Qualtran right now, but this is one of the issue that arises with controlled bloqs so good to keep an eye on.

Reply:
We don't really have a notion of circuit equality in Qualtran. Treating the two control lines of e.g. toffoli as identical is already a problem with or without this PR. This might be a part of #449 roadmap